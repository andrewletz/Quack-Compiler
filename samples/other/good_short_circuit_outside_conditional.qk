class C1() {
    def ShouldNotPrint() : Boolean {
        """THIS SHOULD NOT PRINT\n""".PRINT();
        return true;
    }

    def ShouldPrint() : Boolean {
        """Good\n""".PRINT();
        return true;
    }

    def GetBoolean(in : Boolean) {
        in.PRINT();
        """\n""".PRINT();
    }
}

x = C1();
x.GetBoolean(5==4 and x.ShouldNotPrint());
x.GetBoolean("Hello"=="Hello" and x.ShouldPrint());
x.GetBoolean("Hello"=="World" and x.ShouldNotPrint());
x.GetBoolean("Hello"=="Hello" or x.ShouldNotPrint());

x.GetBoolean("Hello"=="World" or x.ShouldPrint());
