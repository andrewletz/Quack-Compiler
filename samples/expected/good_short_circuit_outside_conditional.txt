false
Good
true
false
true
Good
true
